### DISCUSSION TITLE
enum

### DISCUSSION THREAD

**Discussion by tomByrer** [DanielXMoore/Civet#227] [Ideas]
```ts
enum PizzaTopping { Cheese, Pepperoni, Peppers, Pineapple }
```

I know many hate TS's enums, but I liked them, & I dislike TS in general 😜 
Enums let other programmers know at a glance what are valid inputs.

**Comment by STRd6** [comment]
We should allow enums at least when transpiling to `.ts`. We may need to throw an error in `js` mode until we support equivalent TS enum runtime code.

**Reply by tomByrer** [reply]
Maybe just a warning in JS mode, &/or convert to
```js
let PizzaTopping; //enum Cheese, Pepperoni, Peppers, Pineapple
(or just)
let PizzaTopping;
```
at least short-term.  v2 Civet could add run-time verification....

**Comment by zolomatok** [comment]
+1 for enums, for the reason Tom cited, plus with code completion you can be sure you typed the right thing, and also it'll be properly syntax highlighted.

We can do

`let blah = "something" | "someOtherThing"`

as a technical workaround, but then we don't get any of the above benefits.

**Comment by edemaine** [comment]
Enum should now be supported in Civet 0.5.77, even when outputting JavaScript! Let us know if you run into any bugs or missing cases.

**Reply by zolomatok** [reply]
That's awesome! 🎉
How do you declare an enum though?
Neither of these seem to work
`enum Error = One, Two`
`enum Error = { One, Two }`

**Reply by edemaine** [reply]
Just like in TypeScript:
```ts
enum Error { One, Two }
```
Or use indentation form:
```coffee
enum Error
  One
  Two
```
See also https://civet.dev/cheatsheet#enum

This parallels `class`, `interface`, etc.

**Reply by zolomatok** [reply]
I'm on 0.5.78

**Reply by zolomatok** [reply]
Omg, of course, no `=` required. Thank you!
